pkg/scheduler/framework/plugins/defaultpreemption/default_preemption.go

```diff
@@ -98,9 +98,10 @@ func (pl *DefaultPreemption) PostFilter(ctx context.Context, state *framework.Cy
 // using the nominated resources and the nominated pod could take a long time
 // before it is retried after many other pending pods.
 func preempt(ctx context.Context, fh framework.FrameworkHandle, state *framework.CycleState, pod *v1.Pod, m framework.NodeToStatusMap) (string, error) {
-	cs := fh.ClientSet()
-	// TODO(Huang-Wei): get pod from informer cache instead of API server.
-	pod, err := util.GetUpdatedPod(cs, pod)
+	// It's safe to directly fetch pod here. Because the informer cache has already been
+	// initialized when creating the Scheduler obj, i.e., factory.go#MakeDefaultErrorFunc().
+	// However, tests may need to manually initialize the shared pod informer.
+	pod, err := fh.SharedInformerFactory().Core().V1().Pods().Lister().Pods(pod.Namespace).Get(pod.Name)
```

Comment:
May I ask why we need to get the pod again even from the informer? is it for just update the info of pod in case something changed?

Reply:
yeah, How to make sure the cache has the latest version of pod?

Reply:
The pod passed in is from the schedulingQueue, which may be a dated version, esp. considering that this is a preemption case. However, the API obj in informed cache, IMO in most cases is safe to use, which in theory can be a bit late than the live version, but should be good enough to represent the latest version.

Reply:
got it, thanks